index.html

```diff
@@ -516,6 +517,111 @@ <h2>Algorithms</h2>
+`ecdsa-rdfc-2019` then:
+            <ol class="algorithm">
+              <li>
+Set |cryptosuite|.|transform| to the algorithm in Section
+<a href="#transformation-ecdsa-rdfc-2019"></a>.
+              </li>
+              <li>
+Set |cryptosuite|.|hash| to the algorithm in Section
+<a href="#hashing-ecdsa-rdfc-2019"></a>.
+              </li>
+              <li>
+Set |cryptosuite|.|createProofConfig| to the algorithm in Section
+<a href="#proof-configuration-ecdsa-rdfc-2019"></a>.
+              </li>
+              <li>
+Set |cryptosuite|.|serializeProof| to the algorithm in Section
+<a href="#proof-serialization-ecdsa-rdfc-2019"></a>.
+              </li>
+              <li>
+Set |cryptosuite|.|verifyProof| to the algorithm in Section
+<a href="#proof-verification-ecdsa-rdfc-2019"></a>.
+              </li>
+            </ol>
+          </li>
+          <li>
+If |options|.|type| is `DataIntegrityProof` and |options|.|cryptosuite| is
+`ecdsa-jcs-2019` then:
+            <ol class="algorithm">
+              <li>
+Set |cryptosuite|.|transform| to the algorithm in Section
+<a href="#transformation-ecdsa-jcs-2019"></a>.
+              </li>
+              <li>
+Set |cryptosuite|.|hash| to the algorithm in Section
+<a href="#hashing-ecdsa-jcs-2019"></a>.
+              </li>
+              <li>
+Set |cryptosuite|.|createProofConfig| to the algorithm in Section
+<a href="#proof-configuration-ecdsa-jcs-2019"></a>.
+              </li>
+              <li>
+Set |cryptosuite|.|serializeProof| to the algorithm in Section
+<a href="#proof-serialization-ecdsa-jcs-2019"></a>.
+              </li>
+              <li>
+Set |cryptosuite|.|verifyProof| to the algorithm in Section
+<a href="#proof-verification-ecdsa-jcs-2019"></a>.
+              </li>
+            </ol>
+          </li>
+          <li>
+If |options|.|type| is `DataIntegrityProof` and |options|.|cryptosuite| is
+`ecdsa-sd-2023` then:
+            <ol class="algorithm">
+              <li>
+Set |cryptosuite|.|transform| to the algorithm in Section
+<a href="#base-proof-transformation-ecdsa-sd-2023"></a>.
+              </li>
+              <li>
+Set |cryptosuite|.|hash| to the algorithm in Section
+<a href="#base-proof-hashing-ecdsa-sd-2023"></a>.
+              </li>
+              <li>
+Set |cryptosuite|.|createProofConfig| to the algorithm in Section
+<a href="#base-proof-configuration-ecdsa-sd-2023"></a>.
+              </li>
+              <li>
+Set |cryptosuite|.|serializeProof| to the algorithm in Section
+<a href="#base-proof-serialization-ecdsa-sd-2023"></a>.
+              </li>
+              <li>
+Set |cryptosuite|.|deriveProof| to the algorithm in Section
+<a href="#add-derived-proof-ecdsa-sd-2023"></a>.
+              </li>
+              <li>
+Set |cryptosuite|.|verifyProof| to the algorithm in Section
+<a href="#verify-derived-proof-ecdsa-sd-2023"></a>.
+              </li>
+            </ol>
+          </li>
+          <li>
+Return |cryptosuite|.
```

Comment:
Note that https://w3c.github.io/vc-data-integrity/#dfn-suite-selection-algorithm expects this to return "failure" for values it can't handle, rather than an empty suite.

Reply:
The algorithm was updated to never return a failure. It either returns an empty cryptosuite instance (which doesn't have the functions set) or one that does. The error handling bits were removed to be out of scope in VC-DATA-INTEGRITY as well. This was largely done to simplify the algorithms. The algorithms in VC-DATA-INTEGRITY presume that they're handed a valid cryptosuite instance... if they're not, errors will be thrown. It's up to the implementations to decide how they handle empty cryptosuite instances.

IOW, this feels like implementation-specific stuff that we shouldn't have to spell out to implementers. Unless I'm missing something, I don't think we lose much if we don't return an error here. If you feel strongly about this, we can rework it so it generates an error.